appsdk/triggerfactory_test.go

```diff
@@ -0,0 +1,192 @@
+//
+// Copyright (c) 2020 Technotects
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package appsdk
+
+import (
+	"context"
+	"fmt"
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/common"
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/trigger/http"
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/trigger/messagebus"
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/trigger/mqtt"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-messaging/pkg/types"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"strings"
+	"sync"
+	"testing"
+)
+
+func TestRegisterCustomTrigger_HTTP(t *testing.T) {
+	name := strings.ToTitle(bindingTypeHTTP)
+
+	sdk := AppFunctionsSDK{}
+	err := sdk.RegisterCustomTrigger(name, nil)
+
+	require.NotNil(t, err, "should throw error")
+	require.NotNil(t, err.Error(), fmt.Sprintf("cannot register custom trigger for builtin type (%s)", name))
```

Comment:
Why is this check needed with the one verifying there is an error? Same in other tests.

Reply:
Old habits, for whatever reason whenever I verified errors in tests used to inspect the message too :)

I think I've since realized a better way to do it.

Reply:
Also it should be equal 🤣